src/main/java/se/liu/ida/hefquin/engine/queryplan/logical/impl/LogicalOperatorBase.java

```diff
@@ -0,0 +1,16 @@
+package se.liu.ida.hefquin.engine.queryplan.logical.impl;
+import se.liu.ida.hefquin.engine.queryplan.logical.LogicalOperator;
+
+public abstract class LogicalOperatorBase implements LogicalOperator {
+	private static int counter = 0;  
+	protected final int id;  
+
+    public LogicalOperatorBase() {
+        this.id = counter++;  
+    }
+    
+    @Override
+    public int getID() {
+        return id;
+    }
```

Comment:
My comment about the indentation was meant for the whole file!

(In your editor / IDE you should be able to enable the visibility of whitespace characters. This should make it easier for you to spot the difference between tab-based indentation and space-based indentation.)